src/form/Form.tsx

```diff
@@ -484,22 +490,44 @@ function Form({
     }, 750),
     [activeStep?.id, formRef, validateElements]
   );
+
+  // Debouncing the re-render due to changing values of referenced fields in hide if rules.
+  // Need to rate limit re-renders here for performance reasons.
+  const debouncedHideRuleRender = useCallback(
+    debounce(() => {
+      // Re-render field/element if any field has changed that is
+      //  referenced in the field/element's hide if rule.
+      setRender((render) => !render);
+    }, 500),
+    [setRender, render]
+  );
+
   useEffect(() => {
     return () => {
       debouncedValidate.cancel();
+      debouncedHideRuleRender.cancel();
     };
-  }, [debouncedValidate]);
+  }, [debouncedValidate, debouncedHideRuleRender]);
 
   const updateFieldValues = (newFieldValues: any, rerender = true) => {
     const entries = Object.entries(newFieldValues);
-    const noChange = entries.every(([key, val]) => fieldValues[key] === val);
-    if (noChange) return false;
+    if (!entries.some(([key, val]) => fieldValues[key] !== val)) return false;
 
     const empty = entries.some(([key, val]) => !val || !fieldValues[key]);
     Object.assign(fieldValues, newFieldValues);
+
     // Always rerender from empty state for display purposes
     if (rerender || empty) setRender((render) => !render);
 
+    // If any fields involved in a hideIf have changed, then trigger a re-render to re-evaluate the hide rule
+    if (
+      !entries.some(
```

Comment:
doing !entries.some is equivalent to doing entries.every on the opposite check, which may read clearer